internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -123,19 +123,29 @@ func (h *Host) RemoveContainer(id string) {
 	delete(h.containers, id)
 }
 
-func (h *Host) getContainerLocked(id string) (*Container, error) {
+func (h *Host) GetRunningContainer(id string) (*Container, error) {
+	h.containersMutex.Lock()
+	defer h.containersMutex.Unlock()
+
 	if c, ok := h.containers[id]; !ok {
 		return nil, gcserr.NewHresultError(gcserr.HrVmcomputeSystemNotFound)
 	} else {
+		if c.Status != containerRunning {
+			return nil, gcserr.NewHresultError(gcserr.HrVmcomputeInvalidState)
```

Comment:
Be nice if this could be expanded on with more info, specifically regarding the *why* of it being an invalid state (not running).

Reply:
changed. lmk if you want to reword